### PR TITLE
Outsource parser error messages

### PR DESCRIPTION
**Pull request by repasics**
Modify tools/gen-strings.py to generate error messages.

JerryScript-DCO-1.0-Signed-off-by: Csaba Repasi repasics@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/parser/js/parser-errors.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include "parser-error-messages.inc.h"
+#undef PARSER_ERROR_DEF
+#endif /* JERRY_PARSER */
+  /** @endcond */
+} parser_error_msg_t;
+
+const lit_utf8_byte_t* parser_get_error_utf8 (uint32_t id);
+lit_utf8_size_t parser_get_error_size (uint32_t id);
+
+#endif /* !PARSER_ERRORS_H */
```
Comment (rerobika) [inline jerry-core/parser/js/parser-errors.h:35]: #endif /* PARSER_ERRORS_H */

File: jerry-core/parser/js/parser-errors.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#ifndef PARSER_ERRORS_H
+#define PARSER_ERRORS_H
+
+#include "lit-globals.h"
+
+typedef enum
+{
+/** @cond doxygen_suppress */
+#if JERRY_PARSER
```
Comment (rerobika) [inline jerry-core/parser/js/parser-errors.h:24]: This should be guarded with `JERRY_ERROR_MESSAGES`

**Review by rerobika — CHANGES_REQUESTED** [review]

**Review by rerobika — APPROVED** [review]
LGTM

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**